crates/pallet-erc20-support/src/lib.rs

```diff
@@ -65,6 +70,17 @@ pub mod pallet {
         /// The currency to be exposed as ERC20 token.
         type Currency: Currency<AccountIdOf<Self, I>>;
 
+        /// Allowance balance type.
+        type AllowanceBalance: From<BalanceOf<Self, I>>
+            + Sub<Output = Self::AllowanceBalance>
```

Comment:
```suggestion
            + Sub
```

I think `Sub` actually has a default that would work for use here. Also, why not `CheckedSub`?

Reply:
Unfortunately, `U256` and `U512` don't implement `CheckedSub`.

Reply:
Let's create a tech-debt issue then!